CMakeLists.txt

```diff
@@ -39,7 +39,17 @@ add_definitions (-DGETTEXT_PACKAGE="${GETTEXT_PACKAGE}" -DLOCALEDIR="${CMAKE_INS
 find_package (PkgConfig REQUIRED)
 include (CheckIncludeFile)
 include (FindPkgConfig)
-pkg_check_modules(SERVICE_DEPS REQUIRED glib-2.0>=2.36 gio-2.0>=2.36 x11>=1.6.7 libxklavier>=5.4 libayatana-common>=0.9.3)
+set(DEPS glib-2.0>=2.36 gio-2.0>=2.36 libayatana-common>=0.9.3)
+
+if(EXISTS "${CMAKE_INSTALL_FULL_LIBDIR}/maliit/plugins/liblomiri-keyboard-plugin.so")
+    set(HAS_LOMIRI_KEYBOARD ON)
+    add_definitions(-DHAS_LOMIRI_KEYBOARD)
+else()
+    set(DEPS ${DEPS} x11>=1.6.7 libxklavier>=5.4)
+endif()
```

Comment:
This if-else-clause is wrong IMHO. We want to build the keyboard indicator with both backends, so this should be:

```
# build Lomiri backend if Maliit has the lomiri-keyboard-plugin
if(EXISTS "${CMAKE_INSTALL_FULL_LIBDIR}/maliit/plugins/liblomiri-keyboard-plugin.so")
￼    set(HAS_LOMIRI_KEYBOARD ON)
￼    add_definitions(-DHAS_LOMIRI_KEYBOARD)
endif()

# always build X11 backend
set(DEPS ${DEPS} x11>=1.6.7 libxklavier>=5.4)
```

Furthermore, I have the question if the Lomiri backend support can be built without Maliit's liblomiri-keyboard-plugin.so in place. If so, then the availability of Lomiri keyboard support should be a runtime condition, not a buildtime condition, IMHO.

Reply:
Errm... Your suggestion would force the X11 dependencies on Lomiri. I don't think they would be happy about that.

Reply:
Good point. And, maybe they have those libs anyway, already? From the DEB maintainer's point of view, we need to make it possible to serve both purpose (Lomiri and X11) with the same build of the keyboard indicator.

The only thing, that comes up is loading the backends at runtime dynamically dependent on which backend is needed.

We could then package the backends (some .so files) in separate bin:pkgs where the X11 backend's bin:pkg has the libx* as dependencies whereas the Lomiri backend's bin:pkg does not (but pulls in maalit's lomiri keyboard plugin).

We must not force ourselves to provide two separate / different binary blobs of the same indicator. This opens a can of worms packaging wise long term.